inference-engine/src/gna_plugin/gna_model_serial.cpp

```diff
@@ -166,7 +166,24 @@ void GNAModelSerial::Import(void *basePointer,
         InferenceEngine::OutputsDataMap& outputsDataMap) {
     is.exceptions(std::istream::failbit);
 
+    for (auto inputIndex = 0; inputIndex < modelHeader.nInputs; inputIndex++) {
+        uint32_t nameSize = 0;
+        readNBits<64>(nameSize, is);
+
+        char* inName = new char[nameSize];
```

Comment:
This looks like leek form me. `std::string` take pointer to c-string but it doesn't release it.

Consider to use `std::array` or `std::vector` created on stack for temporary buffer.

Reply:
std::vector is good enough for that case too, you don't really care for the overhead of vector construction since this is not performance critical code

Reply:
In case of smart pointer I'll suggest last example:
https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr#Example